src/main/java/pl/coderslab/charity/service/UserService.java

```diff
@@ -0,0 +1,72 @@
+package pl.coderslab.charity.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.coderslab.charity.entity.User;
+import pl.coderslab.charity.repository.UserRepository;
+
+import javax.servlet.http.HttpSession;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+
+    private static final String LOGGED_USER_EMAIL_ATTRIBUTE = "loggedUserEmail";
+    private static final String LOGGED_USER_ID_ATTRIBUTE = "loggedUserId";
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void saveUser(User user) {
+        if (!user.getPassword().equals(user.getPasswordRepeat())) {
+            throw new IllegalArgumentException("Passwords not the same!");
+        }
+        //TODO
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        userRepository.save(user);
+    }
+
+    public boolean login(HttpSession session, String email, String password) {
+        Optional<User> optionalUser = userRepository.findByEmail(email);
+        if (optionalUser.isPresent()) {
+            User user = optionalUser.get();
```

Comment:
Ogólnie to zamiast robić `isPresent()` i potem `get()` na obiekcie typu `Optional`, można skorzystać z przetwarzania tego lambdami. Użyć jakiegoś `ifPresent()` albo `map()`